E0009 needs to be updated to new format

From: src/test/compile-fail/E0009.rs

A span_note needs to be updated to a span_label.  We may also want to update the span used for the error to the second label.  This would take this:

```
error[E0009]: cannot bind by-move and by-ref in the same pattern
  --> src/test/compile-fail/E0009.rs:15:15
   |
15 |         Some((y, ref z)) => {}, //~ ERROR E0009
   |               ^
   |
note: by-ref binding occurs here
  --> src/test/compile-fail/E0009.rs:15:18
   |
15 |         Some((y, ref z)) => {}, //~ ERROR E0009
   |                  ^^^^^
```

to:

```
error[E0009]: cannot bind by-move and by-ref in the same pattern
  --> src/test/compile-fail/E0009.rs:15:15
   |
15 |         Some((y, ref z)) => {}, //~ ERROR E0009
   |               -  ^^^^^ both by-ref and by-move used
   |               |
   |               by-move pattern here
```

--- a/src/librustc_const_eval/check_match.rs
+++ b/src/librustc_const_eval/check_match.rs
@@ -1120,10 +1120,11 @@ fn check_legality_of_move_bindings(cx: &MatchCheckCtxt,
                 .span_label(p.span, &format!("moves value into pattern guard"))
                 .emit();
         } else if by_ref_span.is_some() {
-            let mut err = struct_span_err!(cx.tcx.sess, p.span, E0009,
-                                           "cannot bind by-move and by-ref in the same pattern");
-            span_note!(&mut err, by_ref_span.unwrap(), "by-ref binding occurs here");
-            err.emit();
+            struct_span_err!(cx.tcx.sess, p.span, E0009,
+                            "cannot bind by-move and by-ref in the same pattern")
+                    .span_label(p.span, &format!("by-move pattern here"))
+                    .span_label(by_ref_span.unwrap(), &format!("both by-ref and by-move used"))
+                    .emit();
         }
     };
 

--- a/src/test/compile-fail/E0009.rs
+++ b/src/test/compile-fail/E0009.rs
@@ -12,7 +12,7 @@ fn main() {
     struct X { x: (), }
     let x = Some((X { x: () }, X { x: () }));
     match x {
-        Some((y, ref z)) => {}, 
+        Some((y, ref z)) => {},
         //~^ ERROR E0009
         //~| NOTE by-move pattern here
         //~| NOTE both by-ref and by-move used

--- a/src/test/compile-fail/E0009.rs
+++ b/src/test/compile-fail/E0009.rs
@@ -12,7 +12,10 @@ fn main() {
     struct X { x: (), }
     let x = Some((X { x: () }, X { x: () }));
     match x {
-        Some((y, ref z)) => {}, //~ ERROR E0009
+        Some((y, ref z)) => {}, 
+        //~^ ERROR E0009
+        //~| NOTE by-move pattern here
+        //~| NOTE both by-ref and by-move used
         None => panic!()
     }
 }

